gnocchi/incoming/redis.py

```diff
@@ -110,13 +118,19 @@ def process_measure_for_metric(self, metric):
         self._client.ltrim(key, item_len + 1, -1)
 
     def iter_on_sacks_to_process(self):
-        self._client.config_set("notify-keyspace-events", "El")
+        self._client.config_set("notify-keyspace-events", "K$")
         p = self._client.pubsub()
         db = self._client.connection_pool.connection_kwargs['db']
-        channel = b"__keyevent@" + str(db).encode() + b"__:rpush"
-        p.subscribe(channel)
+        keyspace = b"__keyspace@" + str(db).encode() + b"__:"
+        pattern = keyspace + self.SACK_PREFIX.encode() + b"*"
```

Comment:
won't this still match with the existing keys? if we have metricA-uuid that gets pushed to sack 5, it will push to incoming128-5:metricA-uuid and it will set incoming128-5?

won't incoming* match with both?

Reply:
No because the `notify-keyspace-events` is set to `K$` and the `$` means it will only send string operations such as `setnx` that we use here. The rest of Redis operations we use are e.g. `rpush` and that's not matched by this filter.

Reply:
i see. i'll test then.